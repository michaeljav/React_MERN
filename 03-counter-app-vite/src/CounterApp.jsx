import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };
  const handleSubstract = () => {
    setCounter(counter - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };
  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+</button>
      <button onClick={handleSubstract}>-</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isrequered,
};

CounterApp.defaultProps = {
  value: 30,
};
