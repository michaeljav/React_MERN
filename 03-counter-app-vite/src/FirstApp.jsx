import { PropTypes } from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <div data-testid='test-title'>{title}</div>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: 'Michael Javier',
  subTitle: 'No hay subt√≠tulo',
  //title: 'No hay titutlo',
};
