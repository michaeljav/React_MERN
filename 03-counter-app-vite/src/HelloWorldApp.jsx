import { PropTypes } from 'prop-types';
const name = 'Michael 2';

export const App = ({ title, subTitle }) => {
  return (
    <>
      <h1>Hello world22 {name}</h1>;
    </>
  );
};

App.propTypes = {
  // title: PropTypes.string,
  title: PropTypes.string.isRequired,
  // subTitle: PropTypes.number.isRequired,
};
