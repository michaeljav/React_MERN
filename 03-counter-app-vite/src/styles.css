html,
body {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Ariel, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}
