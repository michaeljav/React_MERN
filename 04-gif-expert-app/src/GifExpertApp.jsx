import React, { useState } from 'react';
// import { AddCategory } from './components/AddCategory';
// import { GifGrid } from './components/GifGrid';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories((prevCategories) => [newCategory, ...prevCategories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}

      {/* GIF ITEM */}
    </>
  );
};
