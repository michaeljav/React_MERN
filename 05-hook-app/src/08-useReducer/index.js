export * from './TodoApp.jsx';
export * from './TodoList.jsx';
export * from './TodoItem.jsx';
export * from './TodoAdd.jsx';
export * from './todoReducer.js';
