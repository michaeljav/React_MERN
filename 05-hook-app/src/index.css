body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

/* a {
  margin-right: 5px;
} */
.navbar-brand {
  margin-left: 11px;
}
