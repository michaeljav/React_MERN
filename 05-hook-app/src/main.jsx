import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
// import { HooksApp } from './HooksApp';
// import { CounterApp } from './01-useState/CounterApp';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
import { SimpleForms } from './02-useEffect/SimpleForms';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <SimpleForms />
    {/* <CounterWithCustomHook /> */}
    {/* <HooksApp /> */}
    {/* <CounterApp /> */}
  </React.StrictMode>
);
