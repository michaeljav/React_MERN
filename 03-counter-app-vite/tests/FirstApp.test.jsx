import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
  test('debe de hacer match con el snapshot', () => {
    const title = 'Holam michael';
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el TÃ­tulo en h1', () => {
    const title = 'Hola michael';
    const { container, getByText } = render(<FirstApp title={title} />);

    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector('h1');

    expect(h1.innerHTML).toBe(title);
    expect(h1.innerHTML).toContain(title);
  });
});
