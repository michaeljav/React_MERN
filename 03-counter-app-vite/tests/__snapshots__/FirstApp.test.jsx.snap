// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Holam michael
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Michael Javier
  </p>
</div>
`;
